### DISCUSSION TITLE
Unable to apply follow up `migrate dev` when using supabase and shadow db's

### DISCUSSION THREAD

**Discussion by xzilja** [prisma/prisma#17035] [Q&A]
Hey 👋 

I followed supabase's guide on how to use it with prisma https://supabase.com/docs/guides/integrations/prisma and got everything set up and working with shadow database.

Once I run `migrate dev` for the first time, I get expected results with created migration and tables inside my main db. But if I run it again (without making any changes to the schema) I get following error

```
Error: P3006

Migration `20221227163410_init` failed to apply cleanly to the shadow database. 
Error:
db error: ERROR: permission denied for schema public
   0: migration_core::state::DevDiagnostic
             at migration-engine/core/src/state.rs:264
```

Looks like some sort of permission issue on shadow db? But given that it succeeded initially, I am not sure where to start with debuging this. Will appreciate any help and pointers 🙏

**Comment by pantharshit00** [comment]
Hello,

Prisma users your database user to have permission to create a new temporary database in your database server so that we can replay migration history and calculate if migration will be applied cleanly or not. You will need to make sure your database user have that appropriate permission.

**Comment by lh15** [comment]
Anyone have any luck on this?

**Comment by DylanDonkersgoed** [comment]
For me this was happening because the default permissions changed in PostgreSQL 15 and you now need to explicitly grant access to the public schema.

For example, let's say your database is called testproject and your user is called testuser. You need to authenticate as the postgresql admin user (for me this meant launching psql as the postgres user with `sudo -u postgres psql`, it may differ depending on your setup) and then run the following:

```shell
\c testproject
GRANT ALL ON SCHEMA public TO testuser;
```

After that the migrate command should work.

**Comment by GitVex** [comment]
I have a similar issue, though it seems to go a step further and throw me this error:
```
Error: P3006

Migration `20230111141748_first_migration` failed to apply cleanly to the shadow database. 
Error:
db error: ERROR: must be owner of event trigger pgsodium_trg_mask_update
   0: schema_core::state::DevDiagnostic
             at schema-engine/core/src/state.rs:266
```

I've looked at a couple discussions and it seems to be a problem relating to a recent security patch at supabase which revoked the superuser role from the postgres user (https://github.com/orgs/supabase/discussions/9314).

Yet, I can't seem to find a solution for my problem. When trying to alter the event trigger's owner I get rejected because ... well I dont have the superuser. I will open another discussion on the supabase repository as it feels like it may be better suited there, but the similarity to this thread is still there (https://github.com/orgs/supabase/discussions/14257#discussion-5180644)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
